src/middlewared/middlewared/main.py

```diff
@@ -1380,6 +1391,9 @@ def terminate(self):
                 self.logger.trace("Canceling %r", task)
                 task.cancel()
 
+        # We wait for process pool to gracefully shutdown
+        self.__procpool.shutdown(wait=True)
```

Comment:
Nope, `SIGKILL` is good for allowing us to exit and move on with shutting down of middlewared

Reply:
Sorry, I won't take just "Nope" as answer to this

> I've disabled this `atexit` function intentionally because if there is something hanging in process pool, middleware will also hang.

because it's the same problem as you are solving

Reply:
Uh oh. Apologies, i actually meant SIGKILL is good and the use case you have mentioned makes sense. I have made my changes accordingly using `SIGKILL` now